src/libostree/ostree-repo-commit.c

```diff
@@ -2268,6 +2268,179 @@ ostree_repo_write_commit_detached_metadata (OstreeRepo      *self,
   return TRUE;
 }
 
+gboolean
+_ostree_repo_read_commit_compat_signature (OstreeRepo      *self,
+                                           const char      *checksum,
+                                           GBytes         **out_signature,
+                                           GCancellable    *cancellable,
+                                           GError         **error)
+{
+  gboolean ret = FALSE;
+  char signature_path[_OSTREE_LOOSE_PATH_MAX];
+  g_autoptr(GBytes) ret_signature = NULL;
+  GError *temp_error = NULL;
+
+  _ostree_loose_path_with_extension (signature_path, checksum, "sig");
+  ret_signature = ot_file_mapat_bytes (self->objects_dir_fd,
+                                       signature_path, &temp_error);
+  if (!ret_signature)
+    {
+      if (g_error_matches (temp_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
```

Comment:
Nitpick: if `temp_error` was declared with `g_autoptr()`, here you could simply do
```
ret_signature = ot_file_mapat_bytes();
if (temp_error && !g_error_matches (temp_error, G_IO_ERROR_NOT_FOUND))
  // propagate and return FALSE

// transfer and return TRUE
```

Reply:
(same for the other method below)